src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -76,14 +74,18 @@ public AssemblyScannerResults GetScannableAssemblies()
                 {
                     if (!assembly.IsDynamic)
                     {
-                        ScanAssembly(AssemblyPath(assembly), results, processed);
+                        ScanAssembly(assembly, results);
                     }
                 }
             }
 
+            var processed = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
```

Comment:
@danielmarbach I thought about that. The problem is the `processed` dictionary needs to know whether the assembly (or one of it's dependencies) references NSB, which we don't know before.